pkg/kube/translation/apisix_route.go

```diff
@@ -108,40 +108,29 @@ func (t *translator) TranslateRouteV2alpha1(ar *configv2alpha1.ApisixRoute) ([]*
 		if len(part.Match.Paths) < 1 {
 			return nil, nil, errors.New("empty route paths match")
 		}
-		svc, err := t.ServiceLister.Services(ar.Namespace).Get(part.Backend.ServiceName)
-		if err != nil {
-			return nil, nil, err
+		if part.Backend != nil && len(part.Backends) > 1 {
+			return nil, nil, errors.New("backend and backends are exclusive")
 		}
-		svcPort := int32(-1)
-	loop:
-		for _, port := range svc.Spec.Ports {
-			switch part.Backend.ServicePort.Type {
-			case intstr.Int:
-				if part.Backend.ServicePort.IntVal == port.Port {
-					svcPort = port.Port
-					break loop
-				}
-			case intstr.String:
-				if part.Backend.ServicePort.StrVal == port.Name {
-					svcPort = port.Port
-					break loop
-				}
-			}
-		}
-		if svcPort == -1 {
-			log.Errorw("ApisixRoute refers to non-existent Service port",
-				zap.Any("ApisixRoute", ar),
-				zap.String("port", part.Backend.ServicePort.String()),
-			)
-			return nil, nil, err
+		if part.Backend == nil && len(part.Backends) == 0 {
+			return nil, nil, errors.New("no specified backend")
 		}
+		backends := part.Backends
+		backend := part.Backend
+		if len(backends) > 0 {
+			// Use the first backend as the default backend in Route,
+			// others will be configured in traffic-split plugin.
+			backend = backends[0]
```

Comment:
Confused here, `backend ` will be covered if backends is not empty ?

Reply:
No, already check the exclusive relationship before this, when `backends` not empty, `backend` is `nil`.